src/Microsoft.AspNet.Identity.EntityFramework/IdentityDbContext.cs

```diff
@@ -48,62 +48,59 @@ protected IdentityDbContext()
         protected override void OnModelCreating(ModelBuilder builder)
         {
             builder.Entity<TUser>(b =>
-                {
-                    b.HasKey(u => u.Id);
-                    b.HasIndex(u => u.NormalizedUserName).HasName("UserNameIndex");
-                    b.HasIndex(u => u.NormalizedEmail).HasName("EmailIndex");
-                    b.ToTable("AspNetUsers");
-                    b.Property(u => u.ConcurrencyStamp).IsConcurrencyToken();
-
-                    b.Property(u => u.UserName).HasMaxLength(256);
-                    b.Property(u => u.NormalizedUserName).HasMaxLength(256);
-                    b.Property(u => u.Email).HasMaxLength(256);
-                    b.Property(u => u.NormalizedEmail).HasMaxLength(256);
-                    b.HasMany(u => u.Claims).WithOne().HasForeignKey(uc => uc.UserId);
-                    b.HasMany(u => u.Logins).WithOne().HasForeignKey(ul => ul.UserId);
-                    b.HasMany(u => u.Roles).WithOne().HasForeignKey(ur => ur.UserId);
-                });
+            {
+                b.HasKey(u => u.Id);
+                b.HasIndex(u => u.NormalizedUserName).HasName("UserNameIndex");
+                b.HasIndex(u => u.NormalizedEmail).HasName("EmailIndex");
+                b.ToTable("AspNetUsers");
+                b.Property(u => u.ConcurrencyStamp).IsConcurrencyToken();
+
+                b.Property(u => u.UserName).HasMaxLength(256);
+                b.Property(u => u.NormalizedUserName).HasMaxLength(256);
+                b.Property(u => u.Email).HasMaxLength(256);
+                b.Property(u => u.NormalizedEmail).HasMaxLength(256);
+                b.HasMany(u => u.Claims).WithOne().HasForeignKey(uc => uc.UserId).OnDelete(DeleteBehavior.Cascade);
```

Comment:
@divega @HaoK From what I can tell its going to depend on the type of TKey for some cases. If TKey is a non-nullable type, then all relationships must required because TKey is used for FK properties making them all non-nullable. So in this case all relationships should have cascade delete turned on.

Assuming TKey is nullable--say string, then it looks like both User -> Claims and User => Logins will be optional relationships because the FK is not part of the PK and nothing makes the relationship required. So these two will not get cascade delete by default. User -> Roles should still be required because the FK is part of the primary key and therefore must be required.

Therefore, I would expect two of these three relationship to need OnDelete and the other one not to need it.

However, I don't know much about how the model is supposed to work, but it would seem strange to me that the behavior of the model changes based on the type of TKey, so maybe the relationships should always be marked as required rather than letting it depend on the type of TKey.